idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,411 @@
+        @self.Expression(self.flowsheet().config.time, doc="Equation for polytropic enthalpy") 
+        def delta_enth_polytropic_con(b,t):
+            Tout = properties_out[t].temperature
+            Tin = properties_in[t].temperature
+            return b.delta_enth_polytropic[t] == b.delta_enth_actual[t]-b.deltaS[t]*(Tout-Tin)/log(Tout/Tin)     
+        
+        
+        # -------------------------------------------------------------------------------------------------------
+        # design variable constraints
+        @self.Constraint(self.flowsheet().config.time, doc="Maximum allowable impeller tip speed") 
+        def U2max_cons(b,t):
+            phi = b.mass_flow_coeff[t]
+            return b.U2[t] <= sqrt((1984.1*phi**2 - 616.88*phi + 215.97)*0.7*830)
+
+        # @self.Constraint(self.flowsheet().config.time, doc='Maximum Ma number')
+        # def Mamax_cons(b,t):
+        #     # n_stage = 8
+        #     return b.Ma[t] <= -0.202*log(5) + 1.25
+
+        @self.Expression(self.flowsheet().config.time, doc="Thermodynamic power")
+        def fluid_pow(b,t):
+            flow = properties_in[t].flow_mol 
+            return b.delta_enth_actual[t]*flow
+
+        @self.Expression(self.flowsheet().config.time, doc="Shaft power")
+        def elec_pow(b,t):
+            return b.fluid_pow[t]/(b.efficiency_mech * b.eff_drive)
+
+    def initialize(self, state_args={}, outlvl=6, solver='ipopt',
+        optarg={'tol': 1e-6, 'max_iter':1000}):
+        """
+        Initialize the inlet compressor stage model.  This deactivates the
+        specialized constraints, then does the isentropic compressor initialization,
+        then reactivates the constraints and solves.
+
+        Args:
+            state_args (dict): Initial state for property initialization
+            outlvl (int): Amount of output (0 to 3) 0 is lowest
+            solver (str): Solver to use for initialization
+            optarg (dict): Solver arguments dictionary
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = SolverFactory(solver)
+        opt.options = optarg
+
+        flags = self.control_volume.initialize(outlvl=outlvl-1,
+                                       optarg=optarg,
+                                       solver=solver,
+                                       state_args=state_args)
+
+        print('dof for step 1 =', degrees_of_freedom(self))
+        init_log.info_high("Initialization Step 1 Complete.")
+        
+        # sp is what to save to make sure state after init is same as the start
+        #   saves value, fixed, and active state, doesn't load originally free
+        #   values, this makes sure original problem spec is same but initializes
+        #   the values of free vars
+        sp = StoreSpec.value_isfixed_isactive(only_fixed=True)
```

Comment:
Whilst this will guarantee there are no issues, it strikes me that this could just mask underlying issues in the initializations. I would personally prefer to explicitly activate and deactivate constraints and fix and unfix variables, and then check that the state is the same.

Reply:
Hi @andrewlee94 , I am not sure what you suggests here. I supposed that this method allows me to fix all variables if we do not know that they should be fixed and will be unfixed later. If you prefer doing another solution, please let me know. Thanks.

Reply:
The issue I see here is that there could be a mistake in the initialization routine where a constraint is accidentally left deactivated when one of the intermediate solves is called, resulting in an unexpected degree of freedom. This could lead to unexpected values in the model variables even though the solver reports an optimal solution.

My suggestion is that rather than restoring the model state as you are doing here, we rely on the initialization routine itself to restore the state as it goes, and then do a confirmation check as a test to ensure this is the case. That way, if we do accidentally introduce a degree of freedom in the initialization, it will hopefully show up in the test that the initial state is not the same as the final state and we can identify that there is an error in the initialization routine. By forcibly restoring the state however, we hide any of these potential issues.

Reply:
Hi @andrewlee94 : I have changed the initialization approach, deactivate and activate constraints  with fixed/unfixed variables. Can you check the revised version? 
Also, I am not sure how to fix the inlets with this model when it can be connected to an outlet or the another unit. Is there any standard approach for doing this? Thanks.

Reply:
@minhwvu You should call the `initialize` method of the contorl volume of state blocks first. As part of this, you can set the `hold_State=True` argument, which will leave the state variables fixed and return a list of things that needs to be unfixed later.

```
flags = blk.control_volume.initialize(state_args=state_args, hold_state=True)

# the rest of the initialization routine

blk.control_volume.release_state(flags=flags)
```

Reply:
Hi @andrewlee94 : Added those codes. Is there any modification suggested? Thanks.

Reply:
Hi @andrewlee94 : Can you check the model again ? Thanks!